### PR TITLE
UIU-3064 - profile picture FE compression

### PR DESCRIPTION
**Pull request by manvendra-s-rathore**
## Purpose
[UIU-3064](https://folio-org.atlassian.net/jira/software/c/projects/UIU/boards/76?assignee=712020%3A5ca69dba-5ba2-4ee1-be22-97e32eeff056&selectedIssue=UIU-3064) -  Implement client-side compression for Profile Pictures.

## Approach
I have used an open source JavaScript library [compressor.js](https://github.com/fengyuanchen/compressorjs) to perform image compression prior to uploading to backend.

## Screencast
https://github.com/folio-org/ui-users/assets/105021655/182bbc5d-603d-4a2b-92c8-ced407820eed

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,7 @@
 * Non case-sensitive sorting of title column in Open Loans Screen. Refs UIU-2983.
 * Upload Local File for user profile picture. Refs UIU-2974.
 * Don't allow to display or upload a profile picture to a shadow account. Refs UIU-3045.
+* Implement client-side compression for Profile Pictures. Refs UIU-3064.
```
Comment (Terala-Priyanka) [inline CHANGELOG.md:35]: Could you reword it to something like - "Compress selected local file before uploading to server as profile picture"?

File: src/components/EditSections/EditUserInfo/components/EditUserProfilePicture/EditUserProfilePicture.js [hunk 2]
```diff
@@ -156,9 +164,18 @@ const EditUserProfilePicture = ({ profilePictureId, form, personal }) => {
...
+    try {
+      const compressedResult = await new Promise((resolve, reject) => {
+        // eslint-disable-next-line no-new
+        new Compressor(croppedImage,
+          { ...COMPRESSION_OPTIONS, success: resolve, error: reject });
+      });
+      uploadBlob(compressedResult);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.warn(err.message);
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditUserInfo/components/EditUserProfilePicture/EditUserProfilePicture.js:177]: Does the local file modal close when in error scenario? Please check.

**Comment by manvendra-s-rathore** [comment]
Hey @folio-org/fe-tl-reviewers , as Priyanka is on leave, can anyone please review this PR.

**Review by UladzislauKutarkin — APPROVED** [review]

File: src/components/EditSections/EditUserInfo/components/EditUserProfilePicture/EditUserProfilePicture.test.js [hunk 3]
```diff
@@ -161,6 +152,35 @@ describe('Profile Picture', () => {
...
       fireEvent.click(saveAndCloseButton);
     });
+    it('should handle local file upload compression error scnenario', async () => {
+      Compressor.mockImplementationOnce((croppedImage, options) => {
+        return options.error(new Error('compression failed'));
+      });
+      const updateButton = screen.getByTestId('updateProfilePictureDropdown');
+      await userEvent.click(updateButton);
+      const mockImage = new Image();
+      console.log(mockImage, 'mc img');
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditUserInfo/components/EditUserProfilePicture/EditUserProfilePicture.test.js:162]: Please remove this console.log

**Review by Terala-Priyanka — APPROVED** [review]
LGTM, please address the comment and then merge the PR.
